File: backend/internal/dto/study.go
package dto

import "time"

type StudyTopicRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
}

type StudySubtopicRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
}

type StudyCompleteRequest struct {
	Completed bool `json:"completed"`
}

type StudySubtopicResponse struct {
	ID          uint      `json:"id"`
	TopicID     uint      `json:"topicId"`
	Title       string    `json:"title"`
	Description string    `json:"description,omitempty"`
	Completed   bool      `json:"completed"`
	CreatedAt   time.Time `json:"createdAt"`
	UpdatedAt   time.Time `json:"updatedAt"`
}

type StudyTopicResponse struct {
	ID              uint                    `json:"id"`
	Title           string                  `json:"title"`
	Description     string                  `json:"description,omitempty"`
	Subtopics       []StudySubtopicResponse `json:"subtopics"`
	TotalSubtopics  int                     `json:"totalSubtopics"`
	CompletedCount  int                     `json:"completedCount"`
	ProgressPercent int                     `json:"progressPercent"`
	CreatedAt       time.Time               `json:"createdAt"`
	UpdatedAt       time.Time               `json:"updatedAt"`
}

File: frontend/src/types/study.ts
export type StudySubtopic = {
  id: number;
  topicId: number;
  title: string;
  description?: string;
  completed: boolean;
  createdAt: string;
  updatedAt: string;
};

export type StudyTopic = {
  id: number;
  title: string;
  description?: string;
  subtopics: StudySubtopic[];
  totalSubtopics: number;
  completedCount: number;
  progressPercent: number;
  createdAt: string;
  updatedAt: string;
};

export type StudyTopicPayload = {
  title: string;
  description?: string;
};

export type StudySubtopicPayload = {
  title: string;
  description?: string;
};
